activitysim/defaults/models/workplace_location.py

```diff
@@ -56,21 +56,23 @@ def workplace_size_terms(land_use, destination_choice_size_terms):
 def workplace_location_simulate(persons_merged,
                                 zones,
                                 workplace_location_spec,
-                                distance_skim,
+                                skims,
                                 workplace_size_terms):
 
     choosers = persons_merged.to_frame()
     alternatives = zones.to_frame().join(workplace_size_terms.to_frame())
 
-    skims = {
-        "distance": distance_skim
-    }
+    # set the keys for this lookup - in this case there is a TAZ in the choosers
+    # and a TAZ in the alternatives which get merged during interaction
+    skims.set_keys("TAZ", "TAZ_r")
+    # the skims will be available under the name "skims" for any @ expressions
+    locals_d = {"skims": skims}
```

Comment:
Mostly because it's a choice of hardcoding inside the simulation that skims is available under the name "skims"... or the alternative, which is to allow the user to pass in anything they want with any names they prefer (which can now be referred to in the expressions in the CSV file).

This is used more liberally at the link below (in mode choice), where there is a separate skims object available for the trip away from home as the trip back home, and all of the constants which are stored in YAML are added to the scope in as well (and there are plenty of constants in the mode choice).  

But really it's an attempt to keep simple_simulate with as little custom logic as possible.  Although skims are fairly common, they're certainly not used by all models, and it's unfortunate that they have to be passed in at all.  The only thing they're used for is to add the merged dataframe (of choosers and alternatives) back to the skims object.  I think this could be fixed by having a "model_prep" function but this kind of abstraction might be difficult to follow for domain experts.  I find programming to be a constant push-and-pull from abstraction to domain-specificity and the implementation choices are always a question of taste.  Really I could be talked into going either direction (towards abstraction or towards domain specific).

https://github.com/synthicity/activitysim/pull/34/files#diff-800bb6be26eb1bf48d40b8ccf7e48facR106